modules/service/src/main/java/com/devonfw/module/service/common/base/client/async/AbstractAsyncServiceClient.java

```diff
@@ -120,4 +121,26 @@ private static int getLength(Object[] parameters) {
     return parameters.length;
   }
 
+  // START
+  @Override
+  public <R> CompletableFuture<R> call(R result) {
+
+    CompletableFuture<R> future = null;
+    ServiceClientInvocation<S> invocation = getInvocation();
+    try {
+      future = (CompletableFuture<R>) getCompletableFuture(invocation);
+    } catch (Throwable t) {
+      this.errorHandler.accept(t);
+    }
+    return future;
+
+  }
+
+  /**
+   * @param <R>
+   * @param invocation
+   * @return
+   */
+  protected abstract <R> CompletableFuture<R> getCompletableFuture(ServiceClientInvocation<S> invocation);
```

Comment:
IMHO `getCompletableFuture` is somewhat missleading as this is not a "getter". Why dont we just call this method `call` as well?